drivers/regulator/regulator_axp192.c

```diff
@@ -0,0 +1,352 @@
+	uint16_t idx = 0;
+	int ret = 0;
+
+	LOG_INST_DBG(config->log, "voltage = [min=%d, max=%d]", min_uv, max_uv);
+
+	/* set voltage */
+	ret = linear_range_group_get_win_index(config->desc->ranges, config->desc->num_ranges,
+					       min_uv, max_uv, &idx);
+	if (ret != 0) {
+		LOG_INST_ERR(config->log, "No voltage range window could be deteceted");
+		return ret;
+	}
+
+	idx <<= config->desc->vsel_bitpos;
+
+	LOG_INST_DBG(config->log, "[0x%x]=0x%x mask=0x%x", (uint32_t)config->desc->vsel_reg, idx,
+		     (uint32_t)config->desc->vsel_mask);
+	ret = i2c_reg_update_byte_dt(&config->i2c, config->desc->vsel_reg, config->desc->vsel_mask,
+				     (uint8_t)idx);
+	if (ret != 0) {
+		LOG_INST_ERR(config->log, "Failed to set regulator voltage");
+	}
+
+	return ret;
+}
+
+static int axp192_get_voltage(const struct device *dev, int32_t *volt_uv)
+{
+	const struct regulator_axp192_config *config = dev->config;
+	int ret;
+	uint8_t raw_reg;
+
+	/* read voltage */
+	ret = i2c_reg_read_byte_dt(&config->i2c, config->desc->vsel_reg, &raw_reg);
+	if (ret != 0) {
+		return ret;
+	}
+
+	raw_reg = (raw_reg & config->desc->vsel_mask) >> config->desc->vsel_bitpos;
+
+	ret = linear_range_group_get_value(config->desc->ranges, config->desc->num_ranges, raw_reg,
+					   volt_uv);
+
+	return ret;
+}
+
+static int axp192_set_mode(const struct device *dev, regulator_mode_t mode)
+{
+	const struct regulator_axp192_config *config = dev->config;
+	int ret = 0;
+
+	/* setting workmode is only possible for DCDC1-3 */
+	if (config->desc->workmode_reg != 0) {
+
+		if (config->common.initial_mode == AXP192_DCDC_MODE_PWM) {
+			/* configure PWM mode */
+			LOG_INST_DBG(config->log, "PWM mode enabled");
+			ret = i2c_reg_update_byte_dt(&config->i2c, config->desc->workmode_reg,
+						     config->desc->workmode_mask,
+						     config->desc->workmode_pwm_val);
+		} else {
+			/* configure auto mode (default) */
+			ret = i2c_reg_update_byte_dt(&config->i2c, config->desc->workmode_reg,
+						     config->desc->workmode_mask, 0u);
+		}
+
+		if (ret != 0) {
+			LOG_INST_ERR(config->log, "Setting DCDC workmode failed");
+		}
+	}
+
+	return ret;
+}
+
+static int axp192_get_current_limit(const struct device *dev, int32_t *curr_ua)
+{
+	const struct regulator_axp192_config *config = dev->config;
+
+	*curr_ua = config->desc->max_ua;
+
+	return 0;
+}
+
+static struct regulator_driver_api api = {
+	.enable = axp192_enable,
+	.disable = axp192_disable,
+	.count_voltages = axp192_count_voltages,
+	.list_voltage = axp192_list_voltage,
+	.set_voltage = axp192_set_voltage,
+	.get_voltage = axp192_get_voltage,
+	.set_mode = axp192_set_mode,
+	.get_current_limit = axp192_get_current_limit,
+};
+
+static int regulator_axp192_init(const struct device *dev)
+{
+	const struct regulator_axp192_config *config = dev->config;
+	uint8_t enabled_val;
+	bool is_enabled;
+	int ret = 0;
+
+	regulator_common_data_init(dev);
+
+	LOG_INST_DBG(config->log, "voltage=%d", config->common.init_uv);
```

Comment:
please do not access common fields.

Reply:
fixed in [8e8439f](https://github.com/zephyrproject-rtos/zephyr/pull/58809/commits/8e8439f9b9567921f939ea21aabd6e391fbbbe00)